src/esssans/loki/io.py

```diff
@@ -0,0 +1,168 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+"""
+Loading and merging of LoKI data.
+"""
+
+from collections.abc import Iterable
+from functools import reduce
+from typing import Optional
+
+import scipp as sc
+import scippnexus as snx
+
+from ..common import gravity_vector
+from ..types import (
+    FileList,
+    Incident,
+    LoadedFileContents,
+    MonitorType,
+    NexusDetectorName,
+    NexusInstrumentPath,
+    NeXusMonitorName,
+    NexusSampleName,
+    NexusSourceName,
+    RawData,
+    RawMonitor,
+    RunType,
+    TransformationPath,
+    Transmission,
+)
+from .general import default_parameters as params
+
+DETECTOR_BANK_RESHAPING = {
+    params[NexusDetectorName]: lambda x: x.fold(
+        dim='detector_number', sizes=dict(layer=4, tube=32, straw=7, pixel=512)
```

Comment:
Why is the *detector name* a function that folds? Also, doesn't this modify `default_parameters`, which looks like it might break things?

Have you considered using a `-1` somewhere, to be more flexible in terms of future changes?
```suggestion
        dim='detector_number', sizes=dict(layer=4, tube=-1, straw=7, pixel=512)
```

Reply:
> Why is the detector name a function that folds? Also, doesn't this modify default_parameters, which looks like it might break things?

It's a `:` not a `=`, so I'm just using the name of the detector as a key in a new dict. Isn't that ok?

> Have you considered using a -1 somewhere, to be more flexible in terms of future changes?

I think it would be ok to have that at a later stage. Right now, knowing how inconsistent the files we get may be, I'd rather know immediately if something has changed, instead of this silently reshaping to a different size?

Reply:
> > Why is the detector name a function that folds? Also, doesn't this modify default_parameters, which looks like it might break things?
> 
> It's a `:` not a `=`, so I'm just using the name of the detector as a key in a new dict. Isn't that ok?

Right, I misread.

> 
> > Have you considered using a -1 somewhere, to be more flexible in terms of future changes?
> 
> I think it would be ok to have that at a later stage. Right now, knowing how inconsistent the files we get may be, I'd rather know immediately if something has changed, instead of this silently reshaping to a different size?